lib/core/src/recover/handlers/tests/handle_chain_receive_swap_tests.rs

```diff
@@ -0,0 +1,270 @@
+            lbtc_server_lockup_tx_id: Some(create_history_txid("1111", 100)),
+            lbtc_claim_tx_id: None,
+            lbtc_claim_address: None,
+            btc_user_lockup_tx_id: Some(create_history_txid("3333", 102)),
+            btc_user_lockup_address_balance_sat: 0,
+            btc_user_lockup_amount_sat: 100000,
+            btc_refund_tx_id: Some(create_history_txid("4444", 103)), // Confirmed refund
+        };
+
+        // When there's a lockup and confirmed refund tx, it should be Failed
+        assert_eq!(
+            recovered_data.derive_partial_state(Some(100000), None, false, false),
+            Some(PaymentState::Failed)
+        );
+        assert_eq!(
+            recovered_data.derive_partial_state(Some(100000), None, true, false),
+            Some(PaymentState::Failed)
+        );
+
+        // Test with unconfirmed refund
+        let recovered_data = RecoveredOnchainDataChainReceive {
+            lbtc_server_lockup_tx_id: Some(create_history_txid("1111", 100)),
+            lbtc_claim_tx_id: None,
+            lbtc_claim_address: None,
+            btc_user_lockup_tx_id: Some(create_history_txid("3333", 102)),
+            btc_user_lockup_address_balance_sat: 0,
+            btc_user_lockup_amount_sat: 100000,
+            btc_refund_tx_id: Some(create_history_txid("4444", 0)), // Unconfirmed refund
+        };
+
+        // When there's a lockup and unconfirmed refund tx, it should be RefundPending
+        assert_eq!(
+            recovered_data.derive_partial_state(Some(100000), None, false, false),
+            Some(PaymentState::RefundPending)
+        );
+        assert_eq!(
+            recovered_data.derive_partial_state(Some(100000), None, true, false),
+            Some(PaymentState::RefundPending)
+        );
+    }
+
+    #[test]
+    fn test_derive_partial_state_with_btc_lockup_only() {
+        // Test with correct lockup amount
+        let recovered_data = RecoveredOnchainDataChainReceive {
+            lbtc_server_lockup_tx_id: None,
+            lbtc_claim_tx_id: None,
+            lbtc_claim_address: None,
+            btc_user_lockup_tx_id: Some(create_history_txid("3333", 102)),
+            btc_user_lockup_address_balance_sat: 0,
+            btc_user_lockup_amount_sat: 100000,
+            btc_refund_tx_id: None,
+        };
+
+        // Not expired yet - should be Pending
+        assert_eq!(
+            recovered_data.derive_partial_state(Some(100000), None, false, false),
+            Some(PaymentState::Pending)
+        );
+        // Not expired, waiting for fee acceptance
+        assert_eq!(
+            recovered_data.derive_partial_state(Some(100000), None, false, true),
+            Some(PaymentState::WaitingFeeAcceptance)
+        );
+        // Expired - should be Refundable
+        assert_eq!(
+            recovered_data.derive_partial_state(Some(100000), None, true, false),
+            Some(PaymentState::Pending)
```

Comment:
Shouldn't this be expecting `Refundable`?

Reply:
The `btc_user_lockup_address_balance_sat` needs to be 100000 to be refundable

Reply:
Right! So, I guess it depends on which scenario we intend to test, but IMO the most interesting one would be to check if having that value indeed leads to refundable (I don't think having the balance at 0 when there is no refund and the swap is expired can happen in practice) @roeierez

Reply:
The next test actually address the refundable. I left this one as the transactions/balance fetching is not atomic so  in theory we could have incomplete data which means we should keep scanning the swap.